src/validators/ohri-form-validator.ts

```diff
@@ -3,16 +3,32 @@ import { isTrue } from '../utils/boolean-utils';
 
 export const fieldRequiredErrCode = 'field.required';
 export const fieldOutOfBoundErrCode = 'field.outOfBound';
+export const fieldConditionalRequiredErrCode = 'field.conditionalRequired';
+
+const findFieldByIdentifier = (fields: OHRIFormField[], identifier: string): OHRIFormField | undefined => {
+  return fields.find(field => field.id === identifier);
+}
+
+const checkValueAgainstUuids = (field: OHRIFormField | undefined, uuids: string[]): boolean => {
+  if (!field || !field.value || !field.value.value) {
+    return false; 
+  }
+  return uuids.includes(field.value.value);
+}
 
 export const OHRIFieldValidator: FieldValidator = {
-  validate: (field: OHRIFormField, value: any) => {
+  validate: (field: OHRIFormField, value: any, fields: Array<OHRIFormField>) => {
```

Comment:
You want to pass the values map instead like:
```suggestion
  validate: (field: OHRIFormField, value: any, formValues: Record<string, any>) => {
```

You may end up not needing the nested if statement and the new helpers and simply:
```ts

else if (field.required?.type === "conditionalRequired" && formValues[field.required.referenceQuestionId]?.includes(field.required.referenceQuestionAnswers)) {
   return addError(fieldConditionalRequiredErrCode, field.required.message);
}
```
You may have to update the `FieldValidator` interface to reflect the above.